src/utils/saml.ts

```diff
@@ -0,0 +1,27 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as t from "io-ts";
+import { DOMParser } from "xmldom";
+
+const SAMLResponse = t.type({
+  RelayState: t.string,
+  SAMLResponse: t.string
+});
+export type SAMLResponse = t.TypeOf<typeof SAMLResponse>;
+
+export const getSamlIssuer = (body: unknown): string => {
```

Comment:
there's a more functional syntax if you like, in the lines of

```ts
fromEither(SAMLResponse.decode(body))
  .map(_ => Buffer.from(_.SAMLResponse, "base64").toString("utf8"))
  .chain(_ => fromNullable(new DOMParser().parseFromString(_)))
  .chain(_ => fromNullable(_.getElementsByTagName("saml:Issuer").item(0))
  .chain(_ => fromNullable(_.textContent))
  .getOrElse("UNKNOWN")
```

`fromEither` and `fromNullable` come from `fp-ts/lib/Option`

Reply:
Sure, it was amazing. Fixed in dcd6147